adapters/sspbc/sspbc.go

```diff
@@ -0,0 +1,407 @@
+			BidType = openrtb_ext.BidTypeBanner
+
+			if BidExt, ok := a.adSlots[BidId]; ok {
+				var BidIdStored = BidExt.PbSlot
+				bid.ImpID = BidIdStored
+			} else {
+				glog.Errorf("SSPBC: BidExt for this bid.impid not found - %s", BidId)
+			}
+
+			// read additional data from proxy
+			var BidDataExt SsbcResponseExt
+			if err := json.Unmarshal(bid.Ext, &BidDataExt); err != nil {
+				glog.Errorf("SSPBC: cannot unmarshal Bid Ext data")
+				errors = append(errors, err)
+			} else {
+				var adCreationError error
+
+				// Prepare ads (using different methods for banner, native, video)
+
+				// BANNER
+				bid.AdM, adCreationError = a.createBannerAd(bid, BidDataExt, internalRequest, seatBid.Seat)
+
+				if adCreationError != nil {
+					glog.Errorf("SSPBC: cannot format creative")
+					errors = append(errors, err)
+				} else {
+					// append bid to responses
+					bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+						Bid:     &bid,
+						BidType: BidType,
+					})
+				}
+			}
+		}
+	}
+
+	return bidResponse, errors
+}
+
+func (a *SspbcAdapter) createBannerAd(bid openrtb2.Bid, ext SsbcResponseExt, request *openrtb2.BidRequest, seat string) (string, error) {
+	var mcad McAd
+
+	if strings.Contains(bid.AdM, "<!--preformatted-->") {
+		// Banner ad is already formatted
+		return bid.AdM, nil
+	}
+
+	// create McAd payload
+	mcad.Id = request.ID
+	mcad.Seat = seat
+	mcad.SeatBid = make([]openrtb2.SeatBid, 1)
+	mcad.SeatBid[0].Bid = make([]openrtb2.Bid, 1)
+	mcad.SeatBid[0].Bid[0] = bid
+	mcMarshalled, err := json.Marshal(mcad)
+	if err != nil {
+		glog.Errorf("SSPBC: Cannot Marshal mcad!")
+		return bid.AdM, err
+	}
+
+	mcEncoded := base64.URLEncoding.EncodeToString(mcMarshalled)
+
+	bannerData := &TemplatePayload{
+		SiteId:  ext.SiteId,
+		SlotId:  ext.SlotId,
+		AdLabel: ext.AdLabel,
+		PubId:   ext.PublisherId,
+		Page:    request.Site.Page,
+		Referer: request.Site.Ref,
+		McAd:    mcEncoded,
+	}
+
+	/*
+		Prepare banner html, using template file
+
+		Note: Prebidserver bidders have access only to gdpr data in user ext, which is not what we need (as mcad uses prebid.js gdpr format)
+		Therefore, we are not creating window.gdpr. This will force banner creative to execute it's own call to TCF2
+	*/
+
+	var filledTemplate bytes.Buffer
+	if err := a.bannerTemplate.Execute(&filledTemplate, bannerData); err != nil {
+		glog.Errorf("SSPBC: Cannot execute banner template")
+		return bid.AdM, err
+	}
+
+	/*
+		byteTemplate := []byte(filledTemplate.String())
+		fmt.Println(base64.StdEncoding.EncodeToString(byteTemplate)) */
+
+	return filledTemplate.String(), nil
+}
+
+func getImpSize(Imp openrtb2.Imp) string {
+	if Imp.Video != nil {
+		return fmt.Sprintf("%dx%d", Imp.Video.W, Imp.Video.H)
+	}
+
+	if Imp.Banner != nil {
+		sb := strings.Builder{}
+		areaMax := int64(0)
+		sb.WriteString("1x1")
+		for _, sizeI := range Imp.Banner.Format {
+			areaI := sizeI.W * sizeI.H
+			if areaI > areaMax {
+				areaMax = areaI
+				sb.Reset()
```

Comment:
If the end result is just a single size, there is no need to use a strings.Builder. That's only appropriate for a continuous long string.

Reply:
Ok, I've changed this part of code, and replaced unnecessary strings.Builder with a simpler Sprintf